engine/execution/execution_test.go

```diff
@@ -150,89 +169,43 @@ func TestExecutionFlow(t *testing.T) {
 		Once()
 
 	// submit block from consensus node
-	exeNode.IngestionEngine.Submit(conID.NodeID, proposal)
+	err = sendBlock(&exeNode, conID.NodeID, unittest.ProposalFromBlock(&block))
+	require.NoError(t, err)
+
+	// submit the child block from consensus node, which trigger the parent block
+	// to be passed to BlockProcessable
+	err = sendBlock(&exeNode, conID.NodeID, unittest.ProposalFromBlock(&child))
+	require.NoError(t, err)
 
 	require.Eventually(t, func() bool {
+		// when sendBlock returned, ingestion engine might not have processed
+		// the block yet, because the process is async. we have to wait
 		hub.DeliverAll()
 		return receipt != nil
 	}, time.Second*10, time.Millisecond*500)
 
+	// check that the block has been executed.
+	exeNode.AssertHighestExecutedBlock(t, block.Header)
+
 	providerEngine.AssertExpectations(t)
 	verificationEngine.AssertExpectations(t)
 	consensusEngine.AssertExpectations(t)
 }
 
-func TestBlockIngestionMultipleConsensusNodes(t *testing.T) {
```

Comment:
I think we could leave this one to tests in receipts are propagated to multiple nodes